app/src/main/java/com/example/poste/models/User.java

```diff
@@ -132,6 +195,7 @@ public void onResponse(@NonNull Call<ResponseBody> call, @NonNull Response<Respo
                     try {
                         String jsonResponse = response.body().string();
                         Log.d("Response", jsonResponse);
+                        user.folders.clear();
```

Comment:
Great eye finding the duplication bug!

It might be useful to make a copy of the user's folders before this; that way, if the API call fails for some reason, we can set the folders back to what they were before. Otherwise, if the user loses connection at this moment, it may clear out their application but not rebuild it

Reply:
Has been implemented. Unable to have tested this fringe case though.